pylint/lint/pylinter.py

```diff
@@ -66,7 +66,7 @@
     ModuleDescriptionDict,
     Options,
 )
-from pylint.utils import ASTWalker, FileState, LinterStats, utils
+from pylint.utils import ASTWalker, FileState, LinterStats, merge_stats, utils
```

Comment:
Could you explain why we need the merging of stats for the multi-dir config option?

Reply:
Current state is that some stat counters are reset during `linter.open()`, some are reset in `linter.set_current_module` -> `init_single_module()`, and some
 are not reset at all (error, warning etc in 1st group, all counters in stats.by_module in 2nd group, statement in 3rd). It leads to incorrect score calculation when linter (i.e. main checker) is opened per file, or when it is opened after getting asts.
So I decided to reset all possible counters for each new module by creating new LinterStats object in `set_current_module`.

If stats reset is omitted entirely, then another problem arises:
When jobs>1, the same linter object can be used for checking several modules, stats after each module are copied and then merged.
It leads to a situation when some stats are accounted several times in final result (it's checked in `test_subconfigs_score` in my 1st PR).

Explicit stats reset and merge in single process can be avoided, but it will require additional changes in code for parallel checks. I'd suggest to leave it as a possible optimization in another PR.

Reply:
Sorry, you probably already explain it but I don't fully understand. This explanation seems to point to a general issue with stats merging, not something that has to do with multi-directory configs. Or am I misunderstanding you? If it is just a general issue we should tackle it in a separate PR.